### DISCUSSION TITLE
RabbitMQ Stream protocol support

### DISCUSSION THREAD

**Discussion by trialize** [ruby-amqp/bunny#627] [Ideas]
Are there any plans to support Streams?

https://www.rabbitmq.com/streams.html

**Comment by h0jeZvgoxFepBQ2C** [comment]
Yeah, please support Streams!!!1! ❤️

**Comment by carlhoerberg** [comment]
It's already supported, just set the queue argument x-queue-type  to stream, and when you consume you can set the argument x-stream-offset to a value specified in the link above.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Oh ok cool! Was scanning through the documentation, but couldn't find any mention of the streaming feature.
Would be nice to add this to the documentation!

Thanks for the answer! ❤️

**Comment by h0jeZvgoxFepBQ2C** [comment]
@carlhoerberg do you know if there is any javascript wevbrowser-library which supports streams? I know this is not related to the bunny gem, but just asking :)

**Reply by carlhoerberg** [reply]
Yes https://github.com/cloudamqp/amqp-client.js#websockets

**Reply by h0jeZvgoxFepBQ2C** [reply]
Thank you so much!

**Comment by michaelklishin** [comment]
There are no plans to support the stream protocol in Bunny. Our team follows the "one protocol, one client" philosophy. RabbitMQ Streams clients in Java, Go, Python are all separate from their AMQP 0-9-1 or AMQP 1.0 clients.

You can, however, use streams from Bunny today over AMQP 0-9-1 as if it was a queue.

I don't have any immediate plans to work on a Ruby client for RabbitMQ Streams. This is open source software, you are welcome to scratch your own itch and develop one. The [stream protocol is documented in quite a bit of detail](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbitmq_stream/docs/PROTOCOL.adoc).

**Comment by h0jeZvgoxFepBQ2C** [comment]
Hm, I'm not sure if I understand you correctly.

Is there a difference between a (potential) ruby client for rabbitmq streams and bunny (since you can use bunny already with streams by setting mentioned headers above)?

So I don't understand what is missing, since you can use already the bunny gem for streams?

**Reply by michaelklishin** [reply]
I believe this is answered [in RabbitMQ's stream protocol documentation](https://rabbitmq.com/streams.html). You can use streams from Bunny as if they were queues. But they are not, they provide a different set of operations which may be highly desirable for some applications. For that, you need a Ruby client for the RabbitMQ Stream protocol.

**Reply by h0jeZvgoxFepBQ2C** [reply]
Ah ok I see, thank you!
